### DISCUSSION TITLE
error in process filter: eglot--format-markup: Symbol’s function definition is void: text-property-search-forward

### DISCUSSION THREAD

**Discussion by arneaas** [joaotavora/eglot#1201] [Q&A] [answered]
I've installed eglot 1.14. I'm running emacs 27.1 in Ubuntu under WSL.

There seems to be some incompatibility in my setup, and I can't figure out the problem. I'm getting this message in `*Messages*`:
`error in process filter: eglot--format-markup: Symbol’s function definition is void: text-property-search-forward`

Looking up that function I find it in:
`/usr/share/emacs/27.1/lisp/emacs-lisp/text-property-search.el.gz`

However, I see that there is a compat package installed (probably a requirement for eglot) and it has a different version of that function:
`~/.emacs.d/elpa/compat-29.1.4.1/`

but that is not the active version of the function.

By the way, I just updated eglot to 1.14. I _suspect_ this was not a problem in 1.13, but I'm not sure. I don't have 1.13 available anymore, and I can't find the exact release that was 1.13.

**Comment by joaotavora** [comment]
`compat`  is _not_ a requirement for `eglot`.  Maybe it should be, if it would fix things like this.  Does it? Can you `M-x package-install RET compat RET` to pull `compat` to the latest version and it has the function?

Else I think I need to find some alternative to `text-property-search-forward`.

**Reply by arneaas** [reply]
Thanks for the very quick answer! I can't `package-install` `compat`, I get: `no match`, probably because it is already installed. It's listed in `*Packages*`:
`  compat             29.1.4.1      dependency            Emacs Lisp Compatibility Library`

**Comment by joaotavora** [comment, marked as answer]
There's something odd here: the error message says `Symbol’s function definition is void: text-property-search-forward` but you say you found this function in `text-property-search.el.gz`.  How did you "find" it?  

Can you try adding `(require 'text-property-search)` to you configuration somewhere and see if the problem goes away?

**Comment by arneaas** [comment]
`M-x text-property-search-forward` returns:
`text-property-search-forward is an interactive compiled Lisp function
in ‘text-property-search.el’.
(...)`
Then I click on `text-property-search.el` and it takes me to `text-property-search.el.gz`.

Adding `(require 'text-property-search)` to my configuration fixed it, thanks!

**Reply by joaotavora** [reply]
So what's missing is that line in `eglot.el`.  I'll fix it for the next release. Though that won't fix the the Emacs 26.3 compatibility, which is probably also broken.
